### ISSUE TITLE
Android Auto compatibility

### ISSUE THREAD

**Issue by PhiTa08** [segler-alex/RadioDroid#1150]
Hi there. 
I know it's already asked but this great app is really needed now in Android Auto . Please please please do something . Radiodroid2 is far better than open radio . I really love it ! Nice look nice sound ! It's the best !!! Anyways thanks a lot for the fantastic work you made !

**Comment by morckx** [comment]
If you want Android Auto support for RadioDroid now, you can use the latest release of my unofficial fork: https://github.com/morckx/RadioDroid/releases 

Im using Android Auto with RadioDroid since almost 3 years and it's working perfectly.

**Comment by PhiTa08** [comment]
> If you want Android Auto support for RadioDroid now, you can use the latest release of my unofficial fork: https://github.com/morckx/RadioDroid/releases
> 
> Im using Android Auto with RadioDroid since almost 3 years and it's working perfectly.

Thanks 😉 i will try this ! What is the différence between the two apks ?

**Comment by morckx** [comment]
> Thanks wink i will try this ! What is the différence between the two apks ?

The same as with the original RadioDroid, The free version (originally distributed via F-Droid) lacks some non-open-sourced features, such as casting, I think. If in doubt or if anything does not work as expected, I'd use the play  version (originally distributed via the PlayStore).  I think I might not have tested Android Auto with the free version.

**Comment by PhiTa08** [comment]
> > The same as with the original RadioDroid, The free version (originally distributed via F-Droid) lacks some non-open-sourced features, such as casting, I think. If in doubt or if anything does not work as expected, I'd use the play version (originally distributed via the PlayStore). I think I might not have tested Android Auto with the free version.

Sorry for late answer .
So quality sound is the same as "normal" first app I guess ?

**Comment by PhiTa08** [comment]
Errr ! Why I can't install it again ? 😳

**Comment by morckx** [comment]
> > > The same as with the original RadioDroid, The free version (originally distributed via F-Droid) lacks some non-open-sourced features, such as casting, I think. If in doubt or if anything does not work as expected, I'd use the play version (originally distributed via the PlayStore). I think I might not have tested Android Auto with the free version.
> 
> Sorry for late answer . So quality sound is the same as "normal" first app I guess ?

Yes, at least the same.


> Errr ! Why I can't install it again ? flushed

You have to uninstall the regular RadioDroid first, because my unofficial version is signed with different keys.

**Comment by morckx** [comment]
> @morckx Hi, I was looking into a Tunein replacement and found this. Android Auto is something I would really like though. I see you have been working on a PR for this since 2020 (!) #836 but also that Alex has just last week asked you to push some stable stuff (if you have it). It would also be much appreciated from me if you could get AA in here! Thanks in advance for your work here :)

Yes, my fork https://github.com/morckx/RadioDroid (master branch) works with Android Auto (see [Changelog](https://github.com/morckx/RadioDroid/blob/master/CHANGELOG.md#development-snapshot-084900-morckx---2022-10-16) on how to activate it), but I don't have time, currently, to rebase it and make a proper PR.

**Comment by vdbhb59** [comment]
I have made the android auto changes to the code. Kindly wait until it is build and it would work post that.

**Comment by vdbhb59** [comment]
#1086 linked

**Comment by Zambunny** [comment]
I will test the unofficial fork version but I would love to see Android Auto support on the official version too. I like this app.

**Comment by strk** [comment]
> I have made the android auto changes to the code. Kindly wait until it is build and it would work post that.

How long would it take for the working build to get into f-droid ? (And what will it be versiones) ?

I love your app, thanks for sharing it !

**Comment by vdbhb59** [comment]
> > I have made the android auto changes to the code. Kindly wait until it is build and it would work post that.
> 
> How long would it take for the working build to get into f-droid ? (And what will it be versiones) ?
> 
> I love your app, thanks for sharing it !

The last build had the android-auto integrated. I do not quiet know if F-Droid has that removed due to auto being proprietary or likewise. Maybe you can ask in F-Droid in Gitlab/Telegram, and they can confirm.
BTW, all the great work is due to Alex. :)

**Comment by strk** [comment]
Why is this ticket still open if the support is there ? 
F-Droid has 0.86, which seem to be the latest version, is the "integrated support" supposed to show a RadioDroid icon on the auto screen ?

**Comment by GameBurrow** [comment]
> Why is this ticket still open if the support is there ? F-Droid has 0.86, which seem to be the latest version, is the "integrated support" supposed to show a RadioDroid icon on the auto screen ?

Doesn't seem to be the case with latest version on Play Store 0.86 (git 48ac2d4). AA can't control this app, doesn't list it and doesn't recognize it playing.

**Comment by hvdwolf** [comment]
It does not work indeed. Not with F-droid version and neither with the Play Store version. Tested Yesterday and today again. I enabled Developer Options in Android Auto but that doesn't make a difference either.

**Comment by vdbhb59** [comment]
I will go back and check the code once. Probably broken somewhere.

**Comment by BlackDex** [comment]
Same here. Would love to use this app also in the car. Especially since OpenRadio seems to be taken over and full with ads and it's payware now.

Thanks for this btw!

**Comment by Utini2000** [comment]
@vdbhb59 oh my god you are a hero. I was about to look for a replacement since I just got a new car that supports Android Auto but RadioDroid not being AA compatible. It would be a shame to ditch RadioDroid because no other Radio app comes even close to it. I even thought about somehow integrating RadioDroid in another music player that supports but AA but that didn't work either.

For the love of god, please fix this and add Android Auto support :smile: 

Many thanks in advance!

**Comment by vdbhb59** [comment]
Tried several ways, but it seems broken. Also needs a lot of changes in various places. I will revert the changes soon. 🙁

**Comment by Utini2000** [comment]
@vdbhb59 oh crap :( so it seems like this isn't going to happen anytime soon?

**Comment by vdbhb59** [comment]
Trouble is, the changes do not work anymore. When I had made them, I had checked on my system (A10) that time, it was working. Now it is not.

**Comment by denilsonsa** [comment]
These are all the duplicates about Android Auto feature. It may be worth closing the duplicates and keeping a single one open.

* #315
* #436
* #1086
* #1150
